app/src/main/java/com/rightpoint/emojipuzzler/puzzle/PuzzleViewModel.kt

```diff
@@ -0,0 +1,51 @@
+package com.rightpoint.emojipuzzler.puzzle
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.rightpoint.domain.IDomainClient
+import com.rightpoint.domain.models.EmojiPuzzle
+import com.rightpoint.emojipuzzler.EmojiPuzzleApplication
+import kotlinx.coroutines.*
+
+class PuzzleViewModel: ViewModel() {
+
+    private var domain: IDomainClient = EmojiPuzzleApplication.domainClient
+
+    var puzzle: EmojiPuzzle? = null
+
+    var feedBackLiveData: MutableLiveData<String> = MutableLiveData()
+    var currentEmojiPuzzle: MutableLiveData<String> = MutableLiveData()
+    var loading: MutableLiveData<Boolean> = MutableLiveData()
+
+    private val viewModelJob = Job()
+    private val domainScope = CoroutineScope(Dispatchers.IO + viewModelJob)
+
```

Comment:
#? what made you do it this way with coroutines?  Pretty nifty way to do so, just curious.

Reply:
This approach is ideal in order to clean up any running coroutines after moving away from the viewModel.  There is a viewModelScope extension inside "androidx.lifecycle:lifecycle-viewmodel-ktx:x.x.x" that automatically handles onCleared() callback inside ViewModel to ensure any running jobs are cancelled.  This way you wouldn't have to do what I did in every ViewModel you create.  Now as far as choosing .IO over .Main for the dispatcher, that's because I'd like to ensure anything happening inside the domain dependency runs off the main thread and I only post into the ViewModel's LiveData.